deeplake/enterprise/convert_to_libdeeplake.py

```diff
@@ -145,6 +145,11 @@ def dataset_to_libdeeplake(hub2_dataset):
             raise ValueError("In memory datasets are not supported for libdeeplake")
         elif path.startswith("hub://"):
             token = hub2_dataset._token
+            if token is None or token == "" and not hub2_dataset.public:
```

Comment:
Can this check be simplified to 
`if not token and not hub2_dataset.public:
    token = hub2_dataset.client.get_token()
    if not token:
        raise EmptyTokenException`